<?xml version="1.0"?>
<launch>
  <!-- Read me

This launch file is prepared for remote operation of drone by keyboard.  Because of limitations of
docker in terminal emulation, the node teleop_twist_keyboard can't be run directly from this launch
file, but from outside. To run it do the following: Join the solution docker in another terminal
window and run ./subt/scripts/teleop.bash

  -->
  <arg name="robot_name" default="" />
  <param name="robot_name" value="$(arg robot_name)"/>
  <env name="ROSCONSOLE_CONFIG_FILE"  value="$(find robot)/custom_rosconsole.conf"/>

  <!-- Launch your controller nodes -->
  <node name="$(arg robot_name)_proxy" pkg="proxy" type="ros_proxy" args="$(arg robot_name)" output="screen">
	  <!--remap from="$(arg robot_name)/front_scan" to="map_scan"/-->
	  <remap from="$(arg robot_name)/front_rgbd/image_raw/compressed" to="$(arg robot_name)/front/image_raw/compressed"/>
	  <remap from="$(arg robot_name)/cmd_vel" to="$(arg robot_name)/cmd_vel_osgar" />
	  <remap from="$(arg robot_name)/odom" to="/odom_drone"/>
  </node>

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find robotika)/../default.rviz" />

  <node name="drone_height" pkg="robot" type="drone_height.py">
	<remap from="scan_down" to="$(arg robot_name)/bottom_scan" />
	<remap from="scan_up" to="$(arg robot_name)/top_scan" />
	<remap from="cmd_vel_drone" to="$(arg robot_name)/cmd_vel" />
	<remap from="cmd_vel" to="$(arg robot_name)/cmd_vel_osgar"/>
	<remap from="odom" to="$(arg robot_name)/odom" />
  </node> 
  <group ns="rtabmap">
	<node pkg="rtabmap_ros" type="icp_odometry" name="icp_odometry">
        	<param name="subscribe_scan" type="bool"   value="true"/>
                <param name="frame_id"      value="$(arg robot_name)"/>
                <param name="odom_frame_id" value="$(arg robot_name)/odom"/>
                <remap from="odom" to="/$(arg robot_name)/odom" />
                <!--param name="publish_tf" value="false"/-->
                <remap from="scan" to="/$(arg robot_name)/front_scan"/>
                <param name="Icp/PointToPlane"  type="string" value="true"/>
                <param name="Icp/VoxelSize"     type="string" value="0.05"/>
                <param name="Icp/Epsilon"       type="string" value="0.001"/>
                <param name="Icp/PointToPlaneK"  type="string" value="5"/>
                <param name="Icp/PointToPlaneRadius"  type="string" value="0.3"/>
                <param name="Icp/MaxCorrespondenceDistance" type="string" value="0.1"/>
                <param name="Icp/PM"             type="string" value="true"/>
                <param name="Icp/PMOutlierRatio" type="string" value="0.95"/>
                <param name="Odom/Strategy"        type="string" value="0"/>
                <param name="Odom/GuessMotion"     type="string" value="true"/>
                <param name="Odom/ResetCountdown"  type="string" value="1"/>
                <param name="Odom/ScanKeyFrameThr"  type="string" value="0.9"/>
                <param name="RGBD/AngularUpdate"        type="string" value="0.01"/>
                <param name="RGBD/LinearUpdate"         type="string" value="0.01"/>
	</node>
  </group>

</launch>
